### DISCUSSION TITLE
How to expose tcp://localhost:2375 on Github Action windows container?

### DISCUSSION THREAD

**Discussion by abenevaut** [actions/virtual-environments#7539] [Installed software]
i'm looking to expose tcp://localhost:2375 on Github Action inside a windows container to be able to run rspec (to do docker-in-docker stuff) tests to validate the container.

I'm trying to config the docker daemon inside the container like follow https://github.com/abenevaut/msys2/blob/feat/tests/.github/workflows/test.yml#L42
https://github.com/abenevaut/msys2/blob/master/spec/dockerfile_spec.rb#L9

But without succes for the moment, any idea to share ?

I can't find the C:\Users\runneradmin\AppData\Roaming\Docker\settings.json file, so it seems impossible to do the configuration. I'm not sure i can create the file on the same path as my local machine.

I do not have this kind of problem when doing docker tests on linux 
- https://github.com/abenevaut/yafarel/actions/runs/4779925112/jobs/8497321831#step:5:326
- https://github.com/abenevaut/yafarel/blob/php81/tests/spec/dockerfile_ci_spec.rb#L10

**Comment by vpolikarpov-akvelon** [comment]
Hi @abenevaut. Docker daemon configuration on Windows runners is located at `C:\ProgramData\docker\config\daemon.json`. I hope this helps.

**Reply by abenevaut** [reply]
Thanks, i'll dig that with the doc.

https://docs.docker.com/config/daemon/remote-access/#configuring-remote-access-with-daemonjson

**Reply by abenevaut** [reply]
OMG! It's working!

I created a daemon.json clone:
```
{
  "hosts":  [
    "npipe://",
    "tcp://127.0.0.1:2375"
  ]
}
```

moved it and then restart docker with`restart-service *docker*`.

my pipeline: https://github.com/abenevaut/msys2/actions/runs/5045958136/jobs/9050874580
the working code: https://github.com/abenevaut/msys2/pull/2/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R37

Thank you for your help Vasilii ❤️

**Comment by davidkhala** [comment]
BTW, I want to do similar in ubuntu runner, is there any example?

**Reply by abenevaut** [reply]
I did not have to do similar stuff for linux/docker, i only needed to use the socket to run docker in docker.

```
docker run -v /var/run/docker.sock:/var/run/docker.sock
```
